src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -70,12 +72,42 @@ public static synchronized long getQueueFifoTimeSpanMins() {
   /**
    * Set the FIFO queue time span in minutes.
    *
+   * @deprecated
    * @param timeSpanMins new FIFO queue time span in minutes
    */
   public static synchronized void setQueueFifoTimeSpanMins(long timeSpanMins) {
     queueFifoTimeSpanMins = timeSpanMins;
   }
 
+  /**
+   * Set the FIFO queue time span in minutes.
+   *
+   * @param timeSpanMins new FIFO queue time span in minutes
+   * @throws IllegalArgumentException if unable to read a file
+   */
+  public static synchronized void setQueueFifoTimeSpanMinsCheck(long timeSpanMins) {
+    if (timeSpanMins < HistoricalDataConstants.MIN_QUEUE_SPAN_MINS) {
+      throw new IllegalArgumentException("The FifoTimeSpan must not be less than one.");
```

Comment:
I think this should just be added to the existing #setQueueFifoTimeSpanMins() method. The old method shouldn't need to be deprecated, because the behavior you're introducing is just a check of an existing constraint. Nothing should be affected by the change unless it was already misusing the library, so this would require unnecessary changes in existing connectors.

Reply:
resolved